### PR TITLE
feat(replays): Breadcrumb Icon Tab Navigation

### LINKED ISSUE #36803: Timeline Events should be clickable and open the relevant detailed tab
I want to be able to click on the event circle, and/or the content of the tooltip, and be taken into the related tab.

For PageLoad, LCP, Reload events the Network tab should be opened.
- In future we should scroll and highlight the specific row in the network table. This highlight must be made obvious so the user can follow along.

For UserClick events, it makes sense to highlight the element in the replay player... but for that we'd need to scrub the player to the right spot, which is what the breadcrumbs list does. So we might need something else.

### PR DESCRIPTION
**Pull request by danecando**
### Changes
- Adds click handler to breadcrumb icon that opens related tab

Closes #36803 

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: static/app/views/replays/detail/breadcrumbs/breadcrumbItem.tsx [hunk 1]
```diff
@@ -44,6 +46,20 @@ function BreadcrumbItem({
...
+      case 'debug':
+        setActiveTab('network');
+        break;
+      case 'ui':
+        setActiveTab('dom');
+        break;
+      default:
+        setActiveTab('console');
+        break;
+    }
```
Comment (ryan953) [inline static/app/views/replays/detail/breadcrumbs/breadcrumbItem.tsx:61]: is there an error type too we could link?

Comment (danecando) [reply static/app/views/replays/detail/breadcrumbs/breadcrumbItem.tsx:61]: Unfortunately errors aren't coming through on new replays but yeah should it link to console or issues ?

Comment (ryan953) [reply static/app/views/replays/detail/breadcrumbs/breadcrumbItem.tsx:61]: errors should link to Issues i think.

and console still makes a great default. 

cc @jas-kas this is part of that table we were talking about yesterday. what's what and what goes where.

**Review by ryan953 — APPROVED** [review]

**Comment by ryan953** [comment]
What's the correct ticket for this PR? The linked #36803 looks like it's talking about the Timeline instead.

File: static/app/views/replays/detail/breadcrumbs/breadcrumbItem.tsx [hunk 2]
```diff
@@ -44,6 +46,23 @@ function BreadcrumbItem({
     (e: React.MouseEvent<HTMLElement>) => onClick?.(crumb, e),
```
Comment (ryan953) [inline static/app/views/replays/detail/breadcrumbs/breadcrumbItem.tsx:46]: wait, can we combine these click handlers?

Why scope clicks to only the breadcrumb icon

Comment (danecando) [reply static/app/views/replays/detail/breadcrumbs/breadcrumbItem.tsx:46]: > What's the correct ticket for this PR? The linked #36803 looks like it's talking about the Timeline instead.

Err duhh yeah that's the right ticket I just did the wrong thing. I'll add handler for the timeline as well.

> wait, can we combine these click handlers?

Totally I read the icon part specifically as part of the ticket thats why I did it that way

File: static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx [hunk 3]
```diff
@@ -105,16 +107,41 @@ function Event({
...
-        ))}
+        {crumbs.slice(0, totalStackNumber).map((crumb, index) => {
+          return (
+            <IconNode
+              color={colors[index] || crumb.color}
+              key={crumb.id}
+              stack={{totalStackNumber, index}}
+            />
+          );
+        })}
```
Comment (ryan953) [inline static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx:144]: formatting changes made git history harder to read
```suggestion
        {crumbs.slice(0, totalStackNumber).map((crumb, index) => (
          <IconNode
            color={colors[index] || crumb.color}
            key={crumb.id}
            stack={{totalStackNumber, index}}
          />
        ))}
```

File: static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx [hunk 4]
```diff
@@ -105,16 +107,41 @@ function Event({
   // We just need to stack up to 3 times
   const totalStackNumber = Math.min(crumbs.length, 3);
 
+  // If there is only 1 event use the tab navigation handler on the node
+  const nodeClickHandler = () => {
+    if (crumbs.length === 1) {
```
Comment (ryan953) [inline static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx:112]: oh, i didn't think about this case. sucks to have to deal with this, hrm

**Review by ryan953 — APPROVED** [review]

File: static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx (hunk 4 above)
Comment (danecando) [reply static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx:112]: Just let me know what you think. It works fine but also may be a little bit confusing to users

**Pull request merged**